Don't init ardb until a command is run

jcredding opened the pull request:
This changes the cli commands to not init ardb as part of the main
cli and instead wait and let the commands init it themselves. This
makes it so the cli will not error about being unable to load an
ardb db file when using `--help` or `--version`. The cli will still
error if an actual command is used that then initialized ardb.

This also has other benefits of letting the commands customize how
they want to init ardb (establish a connection or not). The
connect, drop and migrate commands all want to establish a
connection because they can assume the db should already exist. The
create command doesn't want to establish a connection ecause it
assumes the db doesn't already exist. Similarly, the generate
migration command doesn't need to establish a connection to a db
to generate a migration file.

This also changes the `connect_db` method on adapters to actually
handle connecting to the db using activerecord's
`establish_connection`. This allows the cli connect command to init
ardb without establishing a connection and then manually try to
connect using the `connect_db` method. As part of this, I changed
the ardb init logic to call the `connect_db` method as well.
Finally, the `connect_db` uses activerecord's connection pool and
`with_connection` helper to ensure that it can checkout a
connection. This was previously done by using the `connection`
method but this has the side-effect of checking out a connection
and never returning it (unless the clear active connections
method is called). By using the `with_connection` method it
provides the same functionality but checks the connection back
into the pool.

Closes #101 

@kellyredding - Ready for review.

kellyredding commented on lib/ardb/cli.rb:
```diff
@@ -106,15 +105,14 @@ def initialize(argv, stdout = nil, stderr = nil)
-          Ardb.init
-          @adapter.connect_db
+          Ardb.init(false)
+          Ardb.adapter.connect_db
```
@jcredding I'm cool with this in spirit, but we'd need to update the `connect_db` method to actually establish a connection.

kellyredding commented:
@jcredding this all looks great 👎 

We discussed a spin-off effort to update the `Migration` class to expect a config to be passed in instead of using the global directly.

jcredding commented:
@kellyredding - I've updated this and modified the connect db logic slightly and added a paragraph to the commit message to explain the changes. Let me know what you think.

kellyredding commented:
@jcredding assuming the CLI command works with no issues when actually used, I'm cool with this.  I really like using the `connect_db` adapter method in the overall init logic - good call there!

👎
